psdisk/psdisk.c

```diff
@@ -359,227 +394,158 @@ static int psdisk_updatePartsList(ptable_partition_t *pHeaders, uint32_t pCnt)
 
 static int psdisk_updateImg(void)
 {
-	uint32_t pCnt;
-	ptable_partition_t *pHeaders;
+	ptable_t *ptable;
+	int err;
 
-	if ((pHeaders = ptable_readPartitions(&pCnt, &psdisk_common.mem)) == NULL) {
-		fprintf(stderr,"The file contains incorrect partition table.\n");
+	ptable = psdisk_readImg();
+	if (ptable == NULL) {
+		fprintf(stderr, "The file contains incorrect partition table.\n");
 		return -1;
 	}
 
-	if (psdisk_updatePartsList(pHeaders, pCnt) < 0) {
-		free(pHeaders);
+	err = psdisk_updatePartsList(ptable);
+	free(ptable);
+	if (err < 0) {
 		return -1;
 	}
 
 	if (psdisk_createImg() < 0) {
-		free(pHeaders);
 		return -1;
 	}
 
-	free(pHeaders);
-
 	return 0;
 }
 
 
 /* Parsing data from user */
 
-static int psdisk_parseToRm(void)
+static int psdisk_parseToRm(const char *arg)
 {
+	struct plist_node_t *node;
 	int i;
-	struct plist_node_t *pNode;
 
 	/* Parse partition name */
-	for (i = 0; i < strlen(optarg); ++i) {
-		if (!isalnum(optarg[i])) {
-			fprintf(stderr,"Invalid partition name - %s.\n", optarg);
-			return -1;
+	for (i = 0; (i < sizeof(node->part.name)) && (arg[i] != '\0'); i++) {
+		if (!isalnum(arg[i])) {
+			break;
```

Comment:
```suggestion
	for (i = 0; i < sizeof(node->part.name); i++) {
		if (!isalnum(arg[i])) {
			break;
```
Same as in `libptable`.

Reply:
Since `isalnum()` check is also performed by `libptable` is decided to limit partition name checks to look for a non-empty null-terminated string with `length < sizeof(part.name)`.